backend/src/main/java/com/benine/backend/AutoPresetCreator.java

```diff
@@ -27,17 +25,28 @@
    * @throws CameraConnectionException when camera cannot be reached.
    * @throws InterruptedException when interupted while waiting for cam to move.
    * @throws TimeoutException if the camera is moving too slow or not at all.
+   * @throws CameraBusyException if the camera is busy.
+   * @throws IOException if exception occurs when creating the preset image.
+   * @throws StreamNotAvailableException if a camera stream cannot be reached.
    */
   public Collection<Preset> createPresets(IPCamera cam)
-          throws CameraConnectionException, InterruptedException, TimeoutException, IOException, StreamNotAvailableException {
+          throws CameraConnectionException, CameraBusyException, InterruptedException,
+          TimeoutException, IOException, StreamNotAvailableException {
+    if (cam.isBusy()) {
+      throw new CameraBusyException("The camera is busy.", cam.getId());
+    }
+    cam.setBusy(true);
     Position camStartPos = cam.getPosition();
 
     ArrayList<Preset> presets = new ArrayList<Preset>();
     for (ZoomPosition pos : generatePositions(cam)) {
+      cam.setBusy(false);
```

Comment:
I just wanted to make the same comment, but Bryan was earlier. I think it also prevents unwanted behavior by not switching between busy and not busy.